extras/ping/ping.go

```diff
@@ -523,15 +543,14 @@ func (p *Pinger) processPacket(recv *packet) error {
 
 	pktUUID, err := p.getPacketUUID(pkt.Data)
 	if err != nil || pktUUID == nil {
-		log.Println("error", err)
-		return err
+		return fmt.Errorf("%w: %s", errBadPacket, err)
 	}
 
 	timestamp := bytesToTime(pkt.Data[:timeSliceLength])
 	pkt.Rtt = receivedAt.Sub(timestamp)
 
 	if !p.Silent {
-		fmt.Printf("reply from %s: icmp_seq=%d ttl=%d time=%.1f ms\n", pkt.SrcIP, pkt.Seq, pkt.Ttl, float64(pkt.Rtt.Microseconds()/1000))
+		fmt.Printf("reply from %s: icmp_seq=%d ttl=%d time=%.1f ms\n", pkt.SrcIP, pkt.Seq, pkt.Ttl, float64(pkt.Rtt.Seconds()*1e3))
```

Comment:
I think `.Seconds` already returns `float64`, so you can drop the cast to `float64`, probably